packages/sdk/src/types/runtime/actor.ts

```diff
@@ -718,6 +719,20 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
         } as ActorLike;
     }
 
+    /**
+     * Prepare outgoing messages
+     * @hidden
+     */
+    public static sanitize(msg: ActorLike): ActorLike;
+    public static sanitize(msg: Partial<ActorLike>): Partial<ActorLike>;
+    public static sanitize(msg: ActorLike | Partial<ActorLike>): ActorLike | Partial<ActorLike> {
+        msg = resolveJsonValues(msg);
+        if (msg.appearance) {
+            msg.appearance = Appearance.sanitize(msg.appearance);
+        }
+        return msg;
+    }

```

Comment:
Actor is an odd place for this to live. Can it just be a utility function?

Reply:
I guess it's fine here.

---
In reply to: [268831340](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/274#discussion_r268831340) [](ancestors = 268831340)